libraries/extractor/src/main/java/androidx/media3/extractor/ts/PesReader.java

```diff
@@ -107,7 +107,8 @@ public final void consume(ParsableByteArray data, @Flags int flags) throws Parse
             Log.w(TAG, "Unexpected start indicator: expected " + payloadSize + " more bytes");
           }
           // Either way, notify the reader that it has now finished.
-          reader.packetFinished();
+          boolean isEndOfInput = (data.limit() == 0);
```

Comment:
@dsparano, I believe we will trigger `packetFinished` with `isEndOfInput` `true`here when we run out of data if network gets too slow, or if live stream generation runs behind. This may result in unwanted playback experience?

Reply:
oh yes, please ignore the comment.